pkg/controllers/state/cluster.go

```diff
@@ -221,10 +250,12 @@ func (c *Cluster) deleteNode(nodeName string) {
 }
 
 // updateNode is called for every node reconciliation
-func (c *Cluster) updateNode(node *v1.Node) {
+func (c *Cluster) updateNode(ctx context.Context, node *v1.Node) error {
 	c.mu.Lock()
 	defer c.mu.Unlock()
-	c.nodes[node.Name] = c.newNode(node)
+	var err error
+	c.nodes[node.Name], err = c.newNode(ctx, node)
```

Comment:
If an error occurs, will we inadvertently set c.nodes[node.Name] to nil? This will cause it to show up when looping over the nodes.

Reply:
Previously newNode() always returned a node object, even if it failed to populate some portion of it.  I could see it going both ways as some semi-failure modes aren't real failures in most cases (e.g. unable to read the CSINode), but for now I've modified it to return nil in the event of error for consistency sake.